src/controllers/postControllers.ts

```diff
@@ -69,6 +69,34 @@ class PostControllers {
     }
   }
 
+  // REACTION ENDPOINTS
+
+  async createReaction(req: Request, res: Response, next: NextFunction) {
+    const { type }: { type: ReactionType } = req.body;
+    const { postId } = req.params;
+    try {
+      await postService.findPostById(postId);
+      abilityService.for(req.user).throwUnlessCan('create', 'Reaction');
+      await reactionService.createReaction(postId, req.user as User, type);
+      return res.status(HttpStatusCodes.CREATED).json({ ok: true });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  // Delete Reaction
+  async deleteReaction(req: Request, res: Response, next: NextFunction) {
+    const { postId } = req.params;
+    const { userId } = req.params;
+    try {
+      await reactionService.findReaction(postId, userId);
+      abilityService.for(req.user).throwUnlessCan('delete', 'Reaction' );
```

Comment:
Here let's save the result of `await reactionService.findReaction(postId, userId);` into a variable called `reaction`, and let's pass this variable to `abilityService` instead of 'Reaction'. This way we can check if the user is allowed to delete that specific reaction, and not just any reaction in general (if you pass the variable instead of 'Reaction' abilityService can check the id).